Examples/Examples/All Examples/AnimateGeoJSONLine.swift

```diff
@@ -0,0 +1,189 @@
+import UIKit
+import MapboxMaps
+import Turf
+import CoreLocation
+
+
+@objc(AnimateGeoJSONLine)
+public class AnimateGeoJSONLine: UIViewController, ExampleProtocol {
+    
+    internal var mapView: MapView!
+    var currentIndex = 1
+    var timer: Timer?
+    var lineLayer: LineLayer!
+    var allCoordinates: [CLLocationCoordinate2D]!
+    var styleManager: StyleManager?
+
+    public var geoJSONLine = (identifier: "routeLine", source: GeoJSONSource())
+
+    
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        
+        mapView = MapView(with: view.bounds, resourceOptions: resourceOptions())
+        mapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        self.view.addSubview(mapView)
+        
+        let centerCoordinate = CLLocationCoordinate2D(latitude: 45.5076, longitude: -122.6736)
+        
+        mapView.cameraManager.setCamera(centerCoordinate: centerCoordinate,
+                                        zoom: 11.0)
+        
+        allCoordinates = coordinates
+        
+        // Allows the delegate to receive information about map events.
+        mapView.on(.mapLoadingFinished) { [weak self] _ in
+            
+            guard let self = self else { return }
+            self.addLine()
+            self.animatePolyline()
+            
+            // The below line is used for internal testing purposes only.
+             self.finish()
+        }
+    }
+    
+    func addLine() {
+        let coordinate = coordinates[currentIndex]
+        // Create a GeoJSON data source.
+        geoJSONLine.source.data = .feature(Feature(geometry: Geometry.lineString(LineString([coordinate]))))
+        
+        var lineLayer = LineLayer(id: "line-layer")
+        lineLayer.filter = Exp(.eq) {
```

Comment:
This filtering isn't necessary, because we're not adding multiple shapes from the `GeoJSONSource`.